azext_iot/product/providers/provider.py

```diff
@@ -0,0 +1,23 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azext_iot.product.providers.auth import AICSAuthentication
+from azext_iot.sdk.product import AICSAPI
+from azext_iot.product.shared import BASE_URL
+from azext_iot.constants import VERSION as cli_version
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def get_sdk(cmd):
+
+    creds = AICSAuthentication(cmd=cmd, base_url=BASE_URL)
+    api = AICSAPI(base_url=BASE_URL, credentials=creds)
+
+    api.config.add_user_agent("azcli/aics/{}".format(cli_version))
```

Comment:
Don't mind if you want to add other user agents, but we have a standard one for the IoT extension https://github.com/Azure/azure-iot-cli-extension/blob/dev/azext_iot/constants.py#L45, used like https://github.com/Azure/azure-iot-cli-extension/blob/dev/azext_iot/sdk/digitaltwins/azure_digital_twins_api.py#L46

Reply:
Adding user agent here is good as is, in fact at some point we should change for the other SDK's, since this approach means less customization to generated code.